base/strings/util.jl

```diff
@@ -513,6 +513,126 @@ function rpad(
     r == 0 ? stringfn(s, p^q) : stringfn(s, p^q, first(p, r))
 end
 
+"""
+    rtrunc(str::AbstractString, w::Int, replace_str::AbstractString = "…")
+
+Truncate `str` to at most `w` characters (see [`textwidth`](@ref)), replacing the last characters
+with `replace_str` if necessary. The default replacement string is "…".
+
+# Examples
+```jldoctest
+julia> s = rtrunc("🍕🍕 I love 🍕", 10)
+"🍕🍕 I lo…"
+
+julia> textwidth(s)
+10
+
+julia> rtrunc("foo", 3)
+"foo"
+```
+
+!!! compat "Julia 1.12"
+    This function was added in Julia 1.12.
+
+See also [`ltrunc`](@ref) and [`ctrunc`](@ref).
+"""
+function rtrunc(str::AbstractString, w::Int, replace_str::AbstractString = "…")
```

Comment:
```suggestion
function rtrunc(str::AbstractString, w::Int, replace_str::Union{AbstractString,AbstractChar} = '…')
```
using a `Char` here should be marginally more efficient.